### PR TITLE
Actix migration

### PR DESCRIPTION
**Pull request by banditopazzo**
# Actix migration

Switched web framework from rocket to actix to standardize to other exein products

## Implementation (Optional)

Replaced web framework and adapted handlers.
Wrapped every diesel request into an actix `web::block` function.

## I have 

- [x] run `cargo fmt`;
- [x] run `cargo clippy`;
- [x] run `cargo test`and all tests pass;
- [x] linked to the originating issue (if applicable).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Kepler is a vulnerability database and lookup store and API currently utilising
 ### Pre-requisites
 
 * `docker`;
-* `rust nightly >= 1.33`;
+* `rust nightly >= 1.58.1`;
```
Comment (JuxhinDB) [inline README.md:25]: It seems that we can remove nightly toolchain entirely with this refactor? :-)

File: src/api/mod.rs [hunk 2]
```diff
@@ -1,30 +1,60 @@
...
-#[get("/")]
-pub fn index() -> &'static str {
-    "pong"
-}
-
-pub fn run() {
+pub fn run() -> Result<Server, anyhow::Error> {
     dotenv().ok();
 
+    let host = "0.0.0.0";
```
Comment (JuxhinDB) [inline src/api/mod.rs:21]: Wouldn't mind these as default but ideally we do not hardocde the host/port. It seems that the current Rocket implementation allows us to pass [`ROCKET_ADDRESS/PORT`](https://github.com/Exein-io/kepler/blob/daa0cb100fa3be71ab4d2c112f38c044943d240b/docker-compose.yml#L28-L29). Could we refactor those to something more generic and default to `0.0.0.0:8000` if missing?

**Review by JuxhinDB — COMMENTED** [review]
Thanks a lot @banditopazzo! This is a great refactor and seems like we can finally move away from `nightly` toolchain. There are just a few tidbits to clean up and it's good to go.

File: src/api/mod.rs (hunk 2 above)
Comment (banditopazzo) [reply src/api/mod.rs]: very basic implementation. a better configuration method should be implemented but not in this pr, opening a new issue

**Review by JuxhinDB — APPROVED** [review]
Good to go!

**Pull request merged**